kore/test/Test/Kore/Step/Simplification/Integration.hs

```diff
@@ -239,6 +241,70 @@ test_simplificationIntegration =
                     , substitution = mempty
                     }
         assertEqualWithExplanation "" expect actual
+    , testCase "function application introduces definedness condition" $ do
```

Comment:
Could you add a comment with the initial configuration, the rules being applied and the expected result? This should make it easier to understand what happens below.